src/autogluon/bench/runbenchmark.py

```diff
@@ -130,6 +126,34 @@ def run_benchmark(
     _dump_configs(benchmark_dir=benchmark.metrics_dir, configs=configs, file_name="configs.yaml")
 
     if configs.get("METRICS_BUCKET", None):
+        framework_name = module_kwargs["run_kwargs"]["framework"].split(":")[0]
+        upload_path = []
```

Comment:
My suggestion is to programmatically update`upload_path`. For example, you can get all keys in the `run_kwargs`, then:
1. iterate through the keys
2. determine if the value of the key is a string
3. if it is string, make it part of `upload_path`. 

This way, you don't have to make changes to this code in case we need to update `run_kwargs` in future, and the code looks cleaner.

Reply:
`run_kwargs` may be different for `AutoMM` but it is the same for `tabular` and `timeseries` as it follows the AMLB format.
Which means that if there are 10 keys in run_kwargs, and AMLB naming convention uses only 4 keys like `constraint`, `task`, `fold` and `framework` then we will only use that and not make the remaining 6 the part of `upload_path` which means iteration may not help.

Reply:
Maybe we can have a list of keys assigned to a variable and iterate through that?

Reply:
Is iteration required when we know what the AMLB standard is?
for example `run_kwargs` for `tabular/timeseries = framework, constraint, task, fold, mode`
and 'run_kwargs' for `AutoMM = framework, constraint, dataset, mode`

we see that `run_kwargs` will be different for `AutoMM` and `tabular/timeseries` which means even if we were to iterate and use the above logic of assigning keys to a variable it would be as follows

`keys = [constraint, framework, fold, mode, dataset, task]` this variable contains keys for both `tabular/ts` and `AutoMM`.
If we now iterate over this in the case of `tabular`, we must make sure that we do not add `dataset` and if it's the case of `AutoMM` then we must not add `fold` and `task`, which makes the logic as `iteration + if conditions`, which in turn will equate to the similar logic as above, maybe even with more LoC.
Since we iterate through the whole of keys list then we will be adding an overhead of `O(n-k)` operations every time, where `k` are the required keys and `n` is the total keys, and `n > k`. Along with the if conditions.
If we do not want the if condition then we would need to make at least two variables called `keys_tabular/ts` and `keys_autoMM` and iterate through each of these two variables which may even increase LoC.

Reply:
I actually meant to have a list inside the if/else, but since there are not many now, I think it's fine to just keep it that way.